scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/PostgresResultsCellRepository.java

```diff
@@ -96,53 +108,185 @@ private static RequestRecord createRequest(
+        final var postCreatedActivitiesAction = new PostCreatedActivitiesAction(connection);
+        final var postSatisfyingActivitiesAction = new PostSatisfyingActivitiesAction(connection)
+    ) {
+      postGoalSatisfactionAction.apply(analysisId, goalSatisfaction);
+      postCreatedActivitiesAction.apply(analysisId, createdActivities);
+      postSatisfyingActivitiesAction.apply(analysisId, satisfyingActivities);
+    }
+  }
+
+  private static ScheduleResults getResults(
+      final Connection connection,
+      final long analysisId
+  ) throws SQLException {
+    try (
+        final var getGoalSatisfactionAction = new GetGoalSatisfactionAction(connection);
+        final var getCreatedActivitiesAction = new GetCreatedActivitiesAction(connection);
+        final var getSatisfyingActivitiesAction = new GetSatisfyingActivitiesAction(connection)
+    ) {
+      final var goalSatisfaction = getGoalSatisfactionAction.get(analysisId);
+      final var createdActivities = getCreatedActivitiesAction.get(analysisId);
+      final var satisfyingActivities = getSatisfyingActivitiesAction.get(analysisId);
+
+      final var goalResults = new HashMap<GoalId, GoalResult>(goalSatisfaction.size());
+      for (final var goalId : goalSatisfaction.keySet()) {
+        goalResults.put(goalId, new GoalResult(
+            createdActivities.get(goalId),
+            satisfyingActivities.get(goalId),
+            goalSatisfaction.get(goalId)
+        ));
+      }
+
+      return new ScheduleResults(goalResults);
     }
   }
 
   public static final class PostgresResultsCell implements ResultsProtocol.OwnerRole {
     private final DataSource dataSource;
-    private final RequestRecord request;
+    private final SpecificationId specId;
+    private final long specRevision;
+    private final long analysisId;
 
     public PostgresResultsCell(
         final DataSource dataSource,
-        final RequestRecord request
+        final SpecificationId specId,
+        final long specRevision,
+        final long analysisId
     ) {
       this.dataSource = dataSource;
-      this.request = request;
+      this.specId = specId;
+      this.specRevision = specRevision;
+      this.analysisId = analysisId;
     }
 
     @Override
     public ResultsProtocol.State get() {
-      // TODO
-      throw new NotImplementedException("PostgresResultsCell not yet implemented");
+      try (final var connection = dataSource.getConnection()) {
+        final var request = getRequest(connection, specId, specRevision);
+        return switch(request.status()) {
+          case "incomplete" -> new ResultsProtocol.State.Incomplete();
+          case "failed" -> new ResultsProtocol.State.Failed(request.failureReason());
+          case "success" -> new ResultsProtocol.State.Success(getResults(connection, request.analysisId()));
+          default -> throw new Error(String.format("Unexpected scheduling request state: %s", request.status()));
```

Comment:
I would rather not parse the string in the action, because I believe the actions are merely for data retrieval. They provide the data that's in the database, but don't operate on it. The interpretation of that data is the responsibility of the relevant postgres repository, and I believe this boundary is beneficial to us (please let me know if your thoughts differ, I only just settled on my understanding of this boundary last week, and am always happy to hear other ideas/suggestions).

Now that said, I agree that having a switch statement on these string values feels less than ideal, but the string-to-State conversion is entirely contained in the `PostgresResultsCellRepository` in both directions*. In other words, the serialization/deserialization using strings for the State of a scheduling request is no more than an implementation detail of the `PostgresResultsCellRepository`.

`ResultsProtocol.State` is an interface (which could be made sealed) with three implementations, each with their own unique associated data. We can't do that with an enum, since all three variants have to have the same structure. We _could_ explore ways to move the strings into the interface implementations themselves, but I think this would cause more complexity than it's worth.

*the `"incomplete"` state is never explicitly set, but is the default value for new rows  added to the `scheduling_request` table

Reply:
I'm cool with whatever you decide on here since at the end of the day it's fairly benign but thought I should give my 2 cents:

>I would rather not parse the string in the action, because I believe the actions are merely for data retrieval. They provide the data that's in the database, but don't operate on it. 

That's generally true but the boundary is not very clear for very discrete types like enums or ID records. For things like `PlanId` we already do parsing within actions, so I don't think an `incomplete`/`failed`/`success` parse is much of a stretch.

> In other words, the serialization/deserialization using strings for the State of a scheduling request is no more than an implementation detail of the PostgresResultsCellRepository.

Agreed, it is an implementation detail but the issue I see is exactly that: the implementation is error prone because we're relying on a custom and case-sensitive string in more than 1 place in the same class. `succeedRequest()` for example:
https://github.com/NASA-AMMOS/aerie/pull/52/files/edd39ea3767ee578e8c0165bb3ffeaddd90f688d#diff-cc4e1dc19800e82575716e95ec48d6a045df8df9af90ab257d7f56dd9293b829R159

>ResultsProtocol.State is an interface (which could be made sealed) with three implementations, each with their own unique associated data. We can't do that with an enum, since all three variants have to have the same structure.

This can be solved with double-dispatch (visitor pattern or `sealed` & `instanceof`).

Reply:
Regarding the boundary between the server code and postgres action code, you're right, in the merlin-server, we build/parse `PlanId` records in postgres actions. We shouldn't. If you look at the new actions for the scheduler-server here, you'll see that I took care to avoid this, because I feel that we should keep our server models and postgres models separate. There are places where it almost feels silly, but keeping this boundary firm allows us to examine either side of it without having to keep the other in mind and reduces coupling between the server code and postgres code to a single location (the repository class itself). I do think if time permits it would actually be worth refactoring the merlin-server postgres actions to conform to this boundary as well.

As for the `ResultProtocol.State` implementation, I'm happy to explore other implementations that enable a more reliable postgres repository implementation, so I'll see what things would look like based on your suggestions and get back to you on that part.

Reply:
Cool, and completely with you on the `PlanId` parsing stuff.

Reply:
@pcrosemurgy Here's the thing about the `ResultsProtocol.State` problem. We need to go from string value to State instance and vice-versa. We could _get_ strings from State objects by adding a method to the State interface, but we can't go in the other direction without basically having a switch statement as we do now. And the fact that we need to compute results for a success variant, or a failure reason for a failed variant makes it even less clear how we would do such a thing.

Reply:
Passing thought: we should consider defining the valid values as a postgres enum, so the database will reject any strings that aren't valid.

Reply:
Ya know, I actually explored that way back when working on merlin-server's simulation_dataset table, but I was still so new to postgres I didn't feel confident in what I was doing and gave up 😅

Reply:
@pcrosemurgy Check out the latest commit on top of this branch. I implemented a Status enum in the RequestRecord to hide the string-comparison details away in a dedicated spot. This was based off some work done with @patkenneally and @mattdailis on the merlin-server yesterday. If desired, I can squash this commit into the relevant commits on the rest of this branch as well.

Reply:
Furthermore, I've added a postgres enum `status_t` for the request status, so the only values it can ever take are the ones we allow (else postgres will give an error).

Reply:
Thanks a lot, this looks great :)